### PR TITLE
[docs] Small fixes to "more about refs"

### PR DESCRIPTION
**Pull request by cody**
* Add a null check to "A Complete Example".
* Fix outdated comment. (Found by @mjomble)
* Replace short dash - with longer dash –.
* Remove backticks from state, because it is not code.

### CODE REVIEW

**Comment by zpao** [comment]
Sorry for missing this, a few comments coming inline…

File: docs/docs/08.1-more-about-refs.md [hunk 1]
```diff
@@ -91,11 +91,13 @@ In order to get a reference to a React component, you can either use `this` to g
 var MyComponent = React.createClass({
   handleClick: function() {
     // Explicitly focus the text input using the raw DOM API.
-    this.myTextInput.focus();
+    if (this.myTextInput !== null) {
+      this.myTextInput.focus();
+    }
```
Comment (zpao) [inline docs/docs/08.1-more-about-refs.md]: This *should* be safe as is but this is safer, so 👍

File: docs/docs/08.1-more-about-refs.md [hunk 2]
```diff
@@ -31,7 +31,7 @@ myComponentInstance.doSomething();
 
 > Note:
 >
-> This should only ever be used at the top level. Inside components, let your `props` and `state` handle communication with child components, or use one of the other methods of getting a ref (string attribute or callbacks).
+> This should only ever be used at the top level. Inside components, let your `props` handle communication with child components, or use one of the other methods of getting a ref (string attribute or callbacks).
```
Comment (zpao) [inline docs/docs/08.1-more-about-refs.md]: Well, you can use your own state to communicate to the children, which is what that's trying to say. You would only do this via the children's props. I see that this isn't clear but I don't think your edit is technically right either.

Comment (cody) [inline docs/docs/08.1-more-about-refs.md]: I have removed this change.

**Comment by zpao** [comment]
Thanks @cody!

**Pull request merged**